sdk/lib/opentelemetry/sdk/trace/tracer.rb

```diff
@@ -11,6 +11,7 @@ module Trace
       class Tracer < OpenTelemetry::Trace::Tracer
         attr_reader :name
         attr_reader :version
+        attr_accessor :instrumentation_library
```

Comment:
We could setup a setter here which would only take name and version. We already have those values from the initializer though, so we could reuse them from there too (the name would then be the full class name then, which isn't awesome IMHO).

Another possibility would be to have a `name` setter in the adapter?